### ISSUE TITLE
Better error responses from Go client compared to Node Client

### ISSUE THREAD

**Issue by davidkel** [hyperledger/fabric-gateway#200]
If I pass in an invalid string for a Certificate or Private Key, in Go I get

- failed to parse certificate PEM
- failed to parse private key PEM

However if I use node then I get for a certificate
- failed to endorse transaction: error validating proposal: access denied: channel [mychannel] creator org unknown, creator is malformed

Which is a response from fabric

and for a private key
- Error: error:0909006C:PEM routines:get_name:no start line

because of calling `crypto.createPrivateKey`

So in the case of Go it's error messages are more helpful using the thin client `identity` facilities, but in node we are subject to a fabric response or a node.js response, both of which are not the most helpful. 

This issue is just to highlight a minor inconsistency between sdks

**Comment by bestbeforetoday** [comment]
I think checking the validity of X.509 certificates (or even that they actually are X.509 certificates!) is beyond the scope of the client API. As far as the client is concerned, the identity is just an opaque chunk of byte data, even in the Go client:
```go
type Identity interface {
	MspID() string       // ID of the Membership Service Provider to which this identity belongs.
	Credentials() []byte // Implementation-specific credentials.
}
```

The client credentials just need to be something that the Fabric nodes understand and verify, and there is no requirement that it is X.509 certificate PEM data. Other signing and identity mechanisms (maybe Idemix?) are free to use whatever credentials they need.

You're just seeing a difference because, in the case of ECDSA key-based security, the credentials are an X.509 certificate PEM and, since Go (and Java) have standard struct (and object) representations of X.509 certificates, we've used those standard object representations. So the errors you're seeing are all in those standard language-specific libraries or in helper functions we've provided to make it easier to create those objects.

In short, I think it's the client application's responsibility to provide suitable credentials, and the server's job to verify those credentials. It would definitely be a bug if you could successfully transact using invalid or unauthorised credentials!

It might be that we can improve the server-generated error messages, although most of the message content is not specific to Fabric Gateway services.

**Comment by bestbeforetoday** [comment]
No plans to change this behaviour within the Fabric Gateway client code.
